admin_manual/enterprise/external_storage/windows-network-drive_configuration.rst

```diff
@@ -154,6 +154,121 @@ In openSUSE, modify the ``/usr/sbin/start_apache2`` file::
 
 Restart Apache, open your ownCloud Admin page and start creating SMB/CIFS mounts.
 
+
+Create Shares programmatically
+------------------------------
+
+Some admins want to create new shares by command line scripts. There are two different approaches.
+
+### 1. occ files_external:create
+
+
+**Personal share**
+
+::
+    occ files_external:create /my_share_name windows_network_drive password::logincredentials --config={host=127.0.0.1,share='home',root='$user',domain='owncloud.local'} --user someuser
+
+**General share**
+
+::
+    occ files_external:create /my_share_name windows_network_drive password::logincredentials --config={host=127.0.0.1,share='home',root='$user',domain='owncloud.local'}
+
+ 
+Arguments
+~~~~~~~~~
+
+> Usage::
+
+
+    files_external:create [options] [--] <mount_point> <storage_backend> <authentication_backend>
+
+::
+
+    Argument | Options
+    -------------|-----------
+    mount point | Path of the mount point within the file system
+    storage_backend | Storage backend indentifier
+    authentication_backend | authentication backend authentifier
+
+    Storage Backend | Identifier
+    ------------------------|-------------
+    Windows Network Drive | windows_network_drive
+    WebDav | dav
+    Local | local
+    ownCloud | owncloud
+    SFTP | sftp
+    Amazon S3 | amazons3
+    Dropbox | dropbox
+    Google Drive | googledrive
+    OpenStack Object Storage | swift
+    SMB / CIFS | smb
+
+    Authentication method | Identifier, name, configuration
+    -----------------------------|-------------
+    Log-in credentials, save in session | password::sessioncredentials 
+    Log-in credentials, save in database | identifier: password::logincredentials 
+    User entered, store in database | password::userprovided *
+    Global Credentials | password::global
+    None | null::null
+    Builtin | builtin::builtin
+    Username and password | password::password
+    OAuth1 | oauth1::oauth1 *
+    OAuth2 | oauth2::oauth2 *
+    RSA public key | publickey::rsa *
+    OpenStack | openstack::openstack *
+    Rackspace | openstack::rackspace *
+    Access key (Amazon S3) | amazons3::accesskey*
+
+
+ \* **authentication methods require aditional configuration**
+**Each Storage Backend needs its corresponding authentication methods.**
+
+There is an other alternative:
+**General Share**
+
+::
+    occ files_external:create /my_share_name windows_network_drive password::logincredentials --config host=127.0.0.1 --config share='home' --config root='$user'  --config domain='somedomain.local' 
+
+**Personal Share**
+
+::
+    occ files_external:create /my_share_name windows_network_drive password::logincredentials --config host=127.0.0.1 --config share='home' --config root='$user'  --config domain='somedomain.local'  --user someuser
+
+
+### 2. occ files_external:import
+The second approach uses a .json file for the whole configuration.
+::
+
+    json
```

Comment:
While there's nothing wrong with embedding JSON, or any other code directly in the documentation, it's better to include it from an external file. That way, you can use native linting and validation tools on the code, which aren't available through the documentation.